core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -477,7 +490,11 @@ public Catalog alterCatalog(NameIdentifier ident, CatalogChange... changes)
   @Override
   public boolean dropCatalog(NameIdentifier ident) {
     // There could be a race issue that someone is using the catalog while we are dropping it.
-    catalogCache.invalidate(ident);
+    CatalogWrapper oldCatalogWrapper = catalogCache.getIfPresent(ident);
+    if (oldCatalogWrapper != null) {
+      catalogCache.invalidate(ident);
+      oldCatalogWrapper.close();
```

Comment:
Can we just do `close` in cache's listener? I found that there're many places doing `close`.

Reply:
That is the point. In the previous version, `close` is automatically executed in close listener, but due to the class loader, we need to judge the corner case (alter operation), in this version, we may the close method manually.